tests/parsers/test_cp.py

```diff
@@ -7,32 +7,46 @@
 from aiida.common import AttributeDict
 
 
-@pytest.fixture
-def generate_inputs(generate_structure):
-    """Return only those inputs that the parser will expect to be there."""
-    return AttributeDict({
-        'structure': generate_structure(),
-        'parameters': orm.Dict(dict={}),
-    })
 
 
+
+#TODO: update the output when the version after 6.5 is out! Now I used a post-6.5 git version (order of atom in the output changed)
+@pytest.mark.parametrize('version',['default','6.5_autopilot','6.5','6.5_cgstep','6.5_cgsteps'])
 def test_cp_default(
```

Comment:
I extended this test. The old one is `test_cp_default` with `version='default`, and it is working.
Is someone more experienced in pytest than me able to tell why two of the others (autopilot and cgstepS) fails with `yaml.representer.RepresenterError: ('cannot represent an object', array([[[ --bands here-- ]]])`? Full output of the test below (the other fails are due to other issues that I have to address)
```
 (aiida-venv) bertossa@pippo:~/sissa/aiida-quantumespresso$ pytest -sv tests/ -k 'test_cp_default'
========================================================================== test session starts ===========================================================================
platform linux -- Python 3.6.9, pytest-5.4.2, py-1.8.1, pluggy-0.13.1 -- /home/bertossa/sissa/aiida-venv/bin/python3
cachedir: .pytest_cache
rootdir: /home/bertossa/sissa/aiida-quantumespresso/tests, inifile: pytest.ini
plugins: regressions-1.0.6, datadir-1.3.1
collected 130 items / 125 deselected / 5 selected                                                                                                                        

tests/parsers/test_cp.py::test_cp_default[default] PASSED
tests/parsers/test_cp.py::test_cp_default[6.5_autopilot] list index out of range
FAILED
tests/parsers/test_cp.py::test_cp_default[6.5] FAILED
tests/parsers/test_cp.py::test_cp_default[6.5_cgstep] list index out of range
FAILED
tests/parsers/test_cp.py::test_cp_default[6.5_cgsteps] list index out of range
FAILED

================================================================================ FAILURES ================================================================================
_____________________________________________________________________ test_cp_default[6.5_autopilot] _____________________________________________________________________

aiida_profile = <aiida.manage.tests.TestManager object at 0x7fa234702a90>, fixture_localhost = <Computer: localhost-test (localhost-test), pk: 1>
generate_calc_job_node = <function generate_calc_job_node.<locals>._generate_calc_job_node at 0x7fa22097d730>
generate_parser = <function generate_parser.<locals>._generate_parser at 0x7fa21ffc7510>
data_regression = <pytest_regressions.data_regression.DataRegressionFixture object at 0x7fa22044d0f0>
generate_structure = <function generate_structure.<locals>._generate_structure at 0x7fa2203f2730>, version = '6.5_autopilot'

    @pytest.mark.parametrize('version',['default','6.5_autopilot','6.5','6.5_cgstep','6.5_cgsteps'])
    def test_cp_default(
        aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression, generate_structure, version
    ):
        """Test a default `cp.x` calculation."""
        entry_point_calc_job = 'quantumespresso.cp'
        entry_point_parser = 'quantumespresso.cp'
        if version=='default':
            def generate_inputs():
                return AttributeDict({
                    'structure': generate_structure(structure_id='silicon'),
                    'parameters': orm.Dict(dict={}),
                })
        else:
            def generate_inputs():
                return AttributeDict({
                    'structure': generate_structure(structure_id='water'),
                    'parameters': orm.Dict(dict={}),
                })
    
        node = generate_calc_job_node(entry_point_calc_job, fixture_localhost, version, generate_inputs())
        parser = generate_parser(entry_point_parser)
        results, calcfunction = parser.parse_from_node(node, store_provenance=False)
    
        assert calcfunction.is_finished, calcfunction.exception
        assert calcfunction.is_finished_ok, calcfunction.exit_message
        assert not orm.Log.objects.get_logs_for(node)
        assert 'output_parameters' in results
        if version is not 'cgstep':
            assert 'output_trajectory' in results
            data_regression.check({
                'parameters': results['output_parameters'].get_dict(),
>               'trajectory': results['output_trajectory'].attributes
            })

tests/parsers/test_cp.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../aiida-venv/lib/python3.6/site-packages/pytest_regressions/data_regression.py:55: in dump
    encoding="utf-8",
../aiida-venv/lib/python3.6/site-packages/yaml/__init__.py:278: in dump_all
    dumper.represent(data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:27: in represent
    node = self.represent_data(data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:48: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:207: in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:118: in represent_mapping
    node_value = self.represent_data(item_value)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:48: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:207: in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:118: in represent_mapping
    node_value = self.represent_data(item_value)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:48: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:207: in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:118: in represent_mapping
    node_value = self.represent_data(item_value)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:58: in represent_data
    node = self.yaml_representers[None](self, data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pytest_regressions.data_regression.RegressionYamlDumper object at 0x7fa220aae710>, data = array([[[-12.07535278,  -6.28316664,  -4.02636205,  -3.17507374]]])

    def represent_undefined(self, data):
>       raise RepresenterError("cannot represent an object", data)
E       yaml.representer.RepresenterError: ('cannot represent an object', array([[[-12.07535278,  -6.28316664,  -4.02636205,  -3.17507374]]]))

../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:231: RepresenterError
__________________________________________________________________________ test_cp_default[6.5] __________________________________________________________________________

aiida_profile = <aiida.manage.tests.TestManager object at 0x7fa234702a90>, fixture_localhost = <Computer: localhost-test (localhost-test), pk: 1>
generate_calc_job_node = <function generate_calc_job_node.<locals>._generate_calc_job_node at 0x7fa22035c8c8>
generate_parser = <function generate_parser.<locals>._generate_parser at 0x7fa21ffc7510>
data_regression = <pytest_regressions.data_regression.DataRegressionFixture object at 0x7fa220aae1d0>
generate_structure = <function generate_structure.<locals>._generate_structure at 0x7fa22035c1e0>, version = '6.5'

    @pytest.mark.parametrize('version',['default','6.5_autopilot','6.5','6.5_cgstep','6.5_cgsteps'])
    def test_cp_default(
        aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression, generate_structure, version
    ):
        """Test a default `cp.x` calculation."""
        entry_point_calc_job = 'quantumespresso.cp'
        entry_point_parser = 'quantumespresso.cp'
        if version=='default':
            def generate_inputs():
                return AttributeDict({
                    'structure': generate_structure(structure_id='silicon'),
                    'parameters': orm.Dict(dict={}),
                })
        else:
            def generate_inputs():
                return AttributeDict({
                    'structure': generate_structure(structure_id='water'),
                    'parameters': orm.Dict(dict={}),
                })
    
        node = generate_calc_job_node(entry_point_calc_job, fixture_localhost, version, generate_inputs())
        parser = generate_parser(entry_point_parser)
        results, calcfunction = parser.parse_from_node(node, store_provenance=False)
    
        assert calcfunction.is_finished, calcfunction.exception
>       assert calcfunction.is_finished_ok, calcfunction.exit_message
E       AssertionError: The stdout output file could not be read.
E       assert False
E        +  where False = <CalcFunctionNode: uuid: 474ab038-9f09-474a-be4b-86bd7493a225 (unstored) (aiida.parsers.parser.parse_calcfunction)>.is_finished_ok

tests/parsers/test_cp.py:39: AssertionError
______________________________________________________________________ test_cp_default[6.5_cgstep] _______________________________________________________________________

aiida_profile = <aiida.manage.tests.TestManager object at 0x7fa234702a90>, fixture_localhost = <Computer: localhost-test (localhost-test), pk: 1>
generate_calc_job_node = <function generate_calc_job_node.<locals>._generate_calc_job_node at 0x7fa21fb94d90>
generate_parser = <function generate_parser.<locals>._generate_parser at 0x7fa21ffc7510>
data_regression = <pytest_regressions.data_regression.DataRegressionFixture object at 0x7fa220a4aeb8>
generate_structure = <function generate_structure.<locals>._generate_structure at 0x7fa21fb94ea0>, version = '6.5_cgstep'

    @pytest.mark.parametrize('version',['default','6.5_autopilot','6.5','6.5_cgstep','6.5_cgsteps'])
    def test_cp_default(
        aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression, generate_structure, version
    ):
        """Test a default `cp.x` calculation."""
        entry_point_calc_job = 'quantumespresso.cp'
        entry_point_parser = 'quantumespresso.cp'
        if version=='default':
            def generate_inputs():
                return AttributeDict({
                    'structure': generate_structure(structure_id='silicon'),
                    'parameters': orm.Dict(dict={}),
                })
        else:
            def generate_inputs():
                return AttributeDict({
                    'structure': generate_structure(structure_id='water'),
                    'parameters': orm.Dict(dict={}),
                })
    
        node = generate_calc_job_node(entry_point_calc_job, fixture_localhost, version, generate_inputs())
        parser = generate_parser(entry_point_parser)
        results, calcfunction = parser.parse_from_node(node, store_provenance=False)
    
        assert calcfunction.is_finished, calcfunction.exception
        assert calcfunction.is_finished_ok, calcfunction.exit_message
>       assert not orm.Log.objects.get_logs_for(node)
E       AssertionError: assert not [<aiida.orm.logs.Log object at 0x7fa21faf5160>]
E        +  where [<aiida.orm.logs.Log object at 0x7fa21faf5160>] = <bound method Log.Collection.get_logs_for of <aiida.orm.logs.Log.Collection object at 0x7fa22073b358>>(<CalcJobNode: uuid: 44058aef-73e1-4097-a73c-986166f9c16e (pk: 18) (aiida.calculations:quantumespresso.cp)>)
E        +    where <bound method Log.Collection.get_logs_for of <aiida.orm.logs.Log.Collection object at 0x7fa22073b358>> = <aiida.orm.logs.Log.Collection object at 0x7fa22073b358>.get_logs_for
E        +      where <aiida.orm.logs.Log.Collection object at 0x7fa22073b358> = <class 'aiida.orm.logs.Log'>.objects
E        +        where <class 'aiida.orm.logs.Log'> = orm.Log

tests/parsers/test_cp.py:40: AssertionError
______________________________________________________________________ test_cp_default[6.5_cgsteps] ______________________________________________________________________

aiida_profile = <aiida.manage.tests.TestManager object at 0x7fa234702a90>, fixture_localhost = <Computer: localhost-test (localhost-test), pk: 1>
generate_calc_job_node = <function generate_calc_job_node.<locals>._generate_calc_job_node at 0x7fa21fa7c950>
generate_parser = <function generate_parser.<locals>._generate_parser at 0x7fa21ffc7510>
data_regression = <pytest_regressions.data_regression.DataRegressionFixture object at 0x7fa21fb45eb8>
generate_structure = <function generate_structure.<locals>._generate_structure at 0x7fa21fa7cd90>, version = '6.5_cgsteps'

    @pytest.mark.parametrize('version',['default','6.5_autopilot','6.5','6.5_cgstep','6.5_cgsteps'])
    def test_cp_default(
        aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression, generate_structure, version
    ):
        """Test a default `cp.x` calculation."""
        entry_point_calc_job = 'quantumespresso.cp'
        entry_point_parser = 'quantumespresso.cp'
        if version=='default':
            def generate_inputs():
                return AttributeDict({
                    'structure': generate_structure(structure_id='silicon'),
                    'parameters': orm.Dict(dict={}),
                })
        else:
            def generate_inputs():
                return AttributeDict({
                    'structure': generate_structure(structure_id='water'),
                    'parameters': orm.Dict(dict={}),
                })
    
        node = generate_calc_job_node(entry_point_calc_job, fixture_localhost, version, generate_inputs())
        parser = generate_parser(entry_point_parser)
        results, calcfunction = parser.parse_from_node(node, store_provenance=False)
    
        assert calcfunction.is_finished, calcfunction.exception
        assert calcfunction.is_finished_ok, calcfunction.exit_message
        assert not orm.Log.objects.get_logs_for(node)
        assert 'output_parameters' in results
        if version is not 'cgstep':
            assert 'output_trajectory' in results
            data_regression.check({
                'parameters': results['output_parameters'].get_dict(),
>               'trajectory': results['output_trajectory'].attributes
            })

tests/parsers/test_cp.py:46: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../aiida-venv/lib/python3.6/site-packages/pytest_regressions/data_regression.py:55: in dump
    encoding="utf-8",
../aiida-venv/lib/python3.6/site-packages/yaml/__init__.py:278: in dump_all
    dumper.represent(data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:27: in represent
    node = self.represent_data(data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:48: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:207: in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:118: in represent_mapping
    node_value = self.represent_data(item_value)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:48: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:207: in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:118: in represent_mapping
    node_value = self.represent_data(item_value)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:48: in represent_data
    node = self.yaml_representers[data_types[0]](self, data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:207: in represent_dict
    return self.represent_mapping('tag:yaml.org,2002:map', data)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:118: in represent_mapping
    node_value = self.represent_data(item_value)
../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:58: in represent_data
    node = self.yaml_representers[None](self, data)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <pytest_regressions.data_regression.RegressionYamlDumper object at 0x7fa21fa2cda0>, data = array([[[0., 0., 0., 0.]]])

    def represent_undefined(self, data):
>       raise RepresenterError("cannot represent an object", data)
E       yaml.representer.RepresenterError: ('cannot represent an object', array([[[0., 0., 0., 0.]]]))

../aiida-venv/lib/python3.6/site-packages/yaml/representer.py:231: RepresenterError
============================================================================ warnings summary ============================================================================
/home/bertossa/sissa/aiida-venv/lib/python3.6/site-packages/past/builtins/misc.py:45
  /home/bertossa/sissa/aiida-venv/lib/python3.6/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import reload

-- Docs: https://docs.pytest.org/en/latest/warnings.html
======================================================================== short test summary info =========================================================================
FAILED tests/parsers/test_cp.py::test_cp_default[6.5_autopilot] - yaml.representer.RepresenterError: ('cannot represent an object', array([[[-12.07535278,  -6.28316664...
FAILED tests/parsers/test_cp.py::test_cp_default[6.5] - AssertionError: The stdout output file could not be read.
FAILED tests/parsers/test_cp.py::test_cp_default[6.5_cgstep] - AssertionError: assert not [<aiida.orm.logs.Log object at 0x7fa21faf5160>]
FAILED tests/parsers/test_cp.py::test_cp_default[6.5_cgsteps] - yaml.representer.RepresenterError: ('cannot represent an object', array([[[0., 0., 0., 0.]]]))
========================================================= 4 failed, 1 passed, 125 deselected, 1 warning in 7.74s =========================================================
(aiida-venv) bertossa@pippo:~/sissa/aiida-quantumespresso$ 

```

@giovannipizzi 
@sphuber 
@ltalirz 
@lorisercole

Reply:
Only native python types can be yaml-serialized, not numpy arrays. Probably there is a way to pass arrays in some way, but the simplest solution is to call the `.tolist()` method of the numpy arrays, so they are converted to lists (or lists of lists if they are 2D), and can be serialised. You are then verifying that the values of the list of lists are the same, but this has the same effect.

Reply:
Ok, I'll try something like this. But were the previous parser's tests parsing any numpy array?

Reply:
The weird thing is that it excepts at `'trajectory': results['output_trajectory'].attributes`, but the attributes of a `TrajectoryData` node *should* be serializable. We do the exact same thing for the other tests. So the attributes must contain something weird, but cannot imagine what. At least not attributes that are set by the class itself. @rikigigi for the test that excepts, can you just print the content of the `results['output_trajectory'].attributes` just before it excepts. This apparently contains non-serializable data, but we need to understand why

Reply:
I added a `pdb.set_trace()` to inspect the structures. The offending arrays are located in output_parameters:
```
(aiida-venv) bertossa@pippo:~/sissa/aiida-quantumespresso/tests$ py.test -k 'test_cp_default'
============================================================================ test session starts ============================================================================
platform linux -- Python 3.6.9, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
rootdir: /home/bertossa/sissa/aiida-quantumespresso/tests, inifile: pytest.ini
plugins: regressions-1.0.6, datadir-1.3.1
collected 130 items / 125 deselected / 5 selected                                                                                                                           

parsers/test_cp.py .
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/bertossa/sissa/aiida-quantumespresso/tests/parsers/test_cp.py(46)test_cp_default()
-> data_regression.check({
(Pdb) pp results['output_parameters'].get_dict()
{'bands': {'bands': array([[[-12.07535278,  -6.28316664,  -4.02636205,  -3.17507374]]]),
           'bands_units': 'eV',
           'occupations': array([[[1., 1., 1., 1.]]])},
 'beta_real_space': False,
 'charge_density': './charge-density.dat',
 'constraint_mag': 0,
 'creator_name': 'cp',
 'creator_version': '6.5',
 'dft_exchange_correlation': None,
 'do_magnetization': False,
 'do_not_use_time_reversal': None,
 'exit_status': 0,
 'fermi_energy_units': 'eV',
 'fft_grid': [60, 60, 60],
 'format_name': 'QEXSD',
 'format_version': '19.03.04',
 'has_dipole_correction': False,
 'has_electric_field': False,
 'inversion_symmetry': False,
 'k_points': [[0.0, 0.0, 0.0]],
 'k_points_units': '1 / angstrom',
 'k_points_weights': [2.0],
 'last_successful_printout': 30,
 'lattice_symmetries': [],
 'lda_plus_u_calculation': False,
 'lkpoint_dir': False,
 'lsda': False,
 'magnetization_angle1': [0.0, 0.0],
 'magnetization_angle2': [0.0, 0.0],
 'no_time_rev_operations': None,
 'non_colinear_calculation': False,
 'number_of_atomic_wfc': 6,
 'number_of_bands': 4,
 'number_of_bravais_symmetries': None,
 'number_of_electrons': 8.0,
 'number_of_k_points': 1,
 'number_of_spin_components': 1,
 'number_of_symmetries': None,
 'q_real_space': False,
 'rho_cutoff': 4353.821352096,
 'rho_cutoff_units': 'eV',
 'smooth_fft_grid': [60, 60, 60],
 'spin_orbit_calculation': False,
 'spin_orbit_domag': False,
 'starting_magnetization': [0.0, 0.0],
 'structure': {'atomic_positions_units': 'Angstrom',
               'atoms': [['H',
                          [0.3077936327955275,
                           0.49074731600773003,
                           0.6020237422809962]],
                         ['H',
                          [-0.25285429616438865,
                           -0.2184308056415271,
                           -0.7702509858867927]],
                         ['O',
                          [-0.16627752338900367,
                           -0.26930737986161374,
                           0.2154517857488999]]],
               'cell': {'atoms': [['H',
                                   [0.3077936327955275,
                                    0.49074731600773003,
                                    0.6020237422809962]],
                                  ['H',
                                   [-0.25285429616438865,
                                    -0.2184308056415271,
                                    -0.7702509858867927]],
                                  ['O',
                                   [-0.16627752338900367,
                                    -0.26930737986161374,
                                    0.2154517857488999]]],
                        'lattice_vectors': [[5.2917720859, 0.0, 0.0],
                                            [0.0, 5.2917720859, 0.0],
                                            [0.0, 0.0, 5.2917720859]],
                        'volume': 148.1847095290449},
               'direct_lattice_vectors_units': 'Angstrom',
               'lattice_parameter': 5.2917720859,
               'lattice_parameter_xml': 10.0,
               'number_of_atoms': 3,
               'number_of_species': 2,
               'reciprocal_lattice_vectors': [[1.0, 0.0, 0.0],
                                              [0.0, 1.0, 0.0],
                                              [0.0, 0.0, 1.0]],
               'species': {'index': [1, 2],
                           'mass': [1.0, 16.0],
                           'pseudo': ['H_HSCV_PBE-1.0.upf',
                                      'O_HSCV_PBE-1.0.upf'],
                           'type': ['H', 'O']}},
 'symmetries': [],
 'symmetries_units': 'crystal',
 'time_reversal_flag': True,
 'wall_time': '     32.51s ',
 'warnings': [],
 'wfc_cutoff': -27.2113834506,
 'wfc_cutoff_units': 'eV'}
(Pdb) pp results['output_trajectory'].attributes
{'array|cell_temperature': [8],
 'array|cells': [8, 3, 3],
 'array|electronic_kinetic_energy': [8],
 'array|energy_constant_motion': [8],
 'array|enthalpy': [8],
 'array|enthalpy_plus_kinetic': [8],
 'array|forces': [0],
 'array|ionic_temperature': [8],
 'array|positions': [8, 3, 3],
 'array|pressure': [8],
 'array|scf_total_energy': [8],
 'array|steps': [8],
 'array|times': [8],
 'array|velocities': [8, 3, 3],
 'array|volume': [8],
 'symbols': ['H', 'H', 'O']}
(Pdb) 
```

Reply:
I added a new terrible fixture to solve the problem (see e2bbb7b)